src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -221,13 +221,23 @@ public boolean isRunning() {
     private final Sensor _recordsDelay;
     private final Sensor _recordsDelayed;
 
-    ConsumeMetrics(final Metrics metrics, final Map<String, String> tags, String topicName) throws ExecutionException, InterruptedException {
-      int partitionCount = 0;
-      DescribeTopicsResult describeTopicsResult = _adminClient.describeTopics(Collections.singleton(topicName));
-      Map<String, KafkaFuture<TopicDescription>> values =  describeTopicsResult.values();
-      partitionCount = values.get(topicName).get().partitions().size();
-      Sensor topicPartitionCount = metrics.sensor("topic-partitions");
-      topicPartitionCount.add(new MetricName("topic-partitions-count", METRIC_GROUP_NAME, "The total number of partitions for the topic.", tags), new Total(partitionCount));
+    ConsumeMetrics(final Metrics metrics, final Map<String, String> tags, String topicName, CompletableFuture<Void> topicPartitionReady) {
```

Comment:
ditto here. The lambda is too long to understand its purpose. Don't call `.thenRun` in the constructor---as long as it is not a constructor! Use it in the caller of the constructor. Really.

Reply:
Discussed in a previous conversation.